buildbot-conf/cordova.conf

```diff
@@ -0,0 +1,326 @@
+            return ['emulator64-x86']
+    return []
+
+def can_find_running_tasks(step):
+    """
+    Return true if an OS and a platform is specified. Those are the
+    criteria for finding a task because running_tasks_on_platform uses
+    those properties to determine which tasks could be running.
+    """
+    return (
+        (step.build.getProperty('slaveos') is not None) and
+        (step.build.getProperty('platform') is not None)
+    )
+
+# renderers
+@renderer
+def render_repo_name(props):
+    repo_url = props.getProperty('repository')
+    return repo_name_from_url(repo_url)
+
+@renderer
+def render_task_kill_command(props):
+
+    os_name       = props.getProperty('slaveos')
+    platform_name = props.getProperty('platform')
+    running_tasks = running_tasks_on_platform(platform_name, os_name)
+
+    if not running_tasks:
+        return ['echo', 'No tasks to kill known.']
+
+    if os_name == WINDOWS:
+        command = ['taskkill', '/F']
+        for task in running_tasks:
+            command.append('/IM')
+            command.append(task)
+
+    else:
+        command = ['killall']
+        command.extend(running_tasks)
+
+    return command
+
+@renderer
+def render_run_args(props):
+    platform = props.getProperty('platform')
+    if platform == 'windows':
+        return '--win'
+    return ''
+
+# step wrappers
+def DescribedStep(step_class, description, haltOnFailure=True, **kwargs):
+    return step_class(description=description, descriptionDone=description, name=slugify(description), haltOnFailure=haltOnFailure, **kwargs)
+
+def SH(workdir=BASE_WORKDIR, timeout=medic_config['app']['timeout'], **kwargs):
+    return DescribedStep(ShellCommand, workdir=workdir, timeout=timeout, **kwargs)
+
+def NPM(npm_command, command=list(), what='code', **kwargs):
+    return SH(command=['npm', npm_command] + command, description='npm ' + npm_command + 'ing ' + what, **kwargs)
+
+def NPMInstall(command=list(), **kwargs):
+    # NOTE:
+    #      adding the --cache parameter so that we don't use the global
+    #      npm cache, which is shared with other processes
+    return NPM('install', command=command + [I('--cache=%(prop:workdir)s/npm_cache')], **kwargs)
+
+def NPMTest(**kwargs):
+    return NPM('test', **kwargs)
+
+def BuildbotClone(repourl, what='code', workdir=None, **kwargs):
+    if workdir is None:
+        workdir = what
+    return DescribedStep(Git, 'cloning ' + what, repourl=repourl, workdir=workdir, mode='full', method='clobber', shallow=True, **kwargs)
+
+def CordovaClone(project_name, **kwargs):
+    branch   = repo_branch_from_name(project_name)
+    repo_url = repo_url_from_name(project_name)
+    return BuildbotClone(repourl=repo_url, branch=branch, what=project_name, **kwargs)
+
+def MedicClone(project_name=None, **kwargs):
+    """
+    Clone repositories using medic's checkout.js script.
+    """
+
+    command = ['node', 'cordova-medic/bin/checkout.js', '--config=cordova-medic/cordova-repos.json']
```

Comment:
I haven't seen any significant slowdowns due to this change; in fact, the clone step is currently three times faster than in the previous config since we clone everything in parallel and at once. I think that this simplifies the code significantly and also makes the config file be the source of all information regarding cloning.